staging/src/k8s.io/client-go/util/workqueue/parallelizer.go

```diff
@@ -50,3 +51,32 @@ func Parallelize(workers, pieces int, doWorkPiece DoWorkPieceFunc) {
 	}
 	wg.Wait()
 }
+
+// ParallelizeUntilFeasible is a very simple framework that allow for parallelizing
+// N independent pieces of work until get feasible solution.
+func ParallelizeUntilFeasible(workers, pieces int, doWorkPiece DoWorkPieceFunc, feasible *int32) {
```

Comment:
Given that this is in client-go, I think we should make this a more generic function. More specifically, I think we should:
1. Rename it to `ParallelizeUntil`.
1. Change `feasible *int32` to a function that returns a bool and an error, similar to [ConditionFunc](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go#L172). When the return value is true, it stops.

Reply:
Thanks for your kind review. :). I've updated it.